libsolidity/analysis/PostTypeChecker.cpp

```diff
@@ -225,13 +225,19 @@ struct OverrideSpecifierChecker: public PostTypeChecker::Checker
 			if (dynamic_cast<ContractDefinition const*>(decl))
 				continue;
 
-			TypeType const* actualTypeType = dynamic_cast<TypeType const*>(decl->type());
+			auto const typeString = [&]()
+			{
+				if (TypeType const* actualTypeType = dynamic_cast<TypeType const*>(decl->type()))
+					return actualTypeType->actualType()->toString(true);
+				else
+					return decl->type()->toString(true);
+			};
 
 			m_errorReporter.typeError(
 				9301_error,
 				override->location(),
 				"Expected contract but got " +
-				actualTypeType->actualType()->toString(true) +
+				typeString() +
```

Comment:
```suggestion
				decl->type()->toString(true) +
```
Why not this?

Reply:
Or ``(actualType ? actualType->actualType() : decl->type())->toString(true)`` with the old definition of ``actualType``? Not that I don't like lambdas, but maybe slight overkill for this :-).
``decl->type()->toString(true)`` actually would also really be fine - in case of a ``TypeType`` it'll just be ``type(<type>)`` instead of ``<type>``, which wouldn't bother at least me.